### PR TITLE
chore: migrate tooling over to .NET 6

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-dotnet/issues/1974
Relates https://github.com/microsoft/playwright-dotnet/pull/1884

Motivation: .NET 5 reaches EOL in May 8, 2022 since it was no LTS, LTS is .NET 6. There are no changes in the TFM so no user facing changes and only tooling.

### CODE REVIEW

File: src/.editorconfig [hunk 1]
```diff
@@ -185,6 +185,7 @@ dotnet_diagnostic.SA1202.severity = none
 dotnet_diagnostic.SA1503.severity = none
 dotnet_diagnostic.RCS1075.severity = none
 dotnet_diagnostic.SA1402.severity = none
+dotnet_diagnostic.CA1844.severity = none
```
Comment (mxschmitt) [inline src/.editorconfig:188]: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1844

Was ocurring here: https://github.com/microsoft/playwright-dotnet/blob/4f52f655bb45143d494848508979067c4654b023/src/Playwright/Core/Stream.cs#L79-L84

File: .build/azure-pipeline-nuget.yml [hunk 2]
```diff
@@ -28,10 +28,10 @@ stages:
     steps:
 
     - task: UseDotNet@2
-      displayName: 'Use .NET Core sdk'
+      displayName: 'Use .NET 6 SDK'
       inputs:
         packageType: sdk
-        version: 5.x
+        version: 6.x
```
Comment (mxschmitt) [inline .build/azure-pipeline-nuget.yml:34]: I think we need to install on our deployment pipeline 5 + 6.

File: src/Playwright/Core/Artifact.cs [hunk 3]
```diff
@@ -69,11 +69,13 @@ public async Task SaveAsAsync(string path)
                 return;
             }
             System.IO.Directory.CreateDirectory(Path.GetDirectoryName(path));
-            await using var stream = await _channel.SaveAsStreamAsync().ConfigureAwait(false);
-
-            using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
+            var stream = await _channel.SaveAsStreamAsync().ConfigureAwait(false);
+            await using (stream)
```
Comment (campersau) [inline src/Playwright/Core/Artifact.cs:73]:
```suggestion
            await using (stream.ConfigureAwait(false))
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -49,14 +51,13 @@ jobs:
...
+      - name: Setup .NET
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.x
-      - name: Setup .NET 5.0 # needed for our build steps
-        uses: actions/setup-dotnet@v1
-        with:
-          dotnet-version: 5.0.x
+          dotnet-version: |
+            3.1.x
```
Comment (kblok) [inline .github/workflows/tests.yml:58]: I would drop 3.1

**Review by pavelfeldman — APPROVED** [review]

File: src/Playwright/Core/Artifact.cs [hunk 5]
```diff
@@ -69,11 +69,13 @@ public async Task SaveAsAsync(string path)
...
             }
             System.IO.Directory.CreateDirectory(Path.GetDirectoryName(path));
-            await using var stream = await _channel.SaveAsStreamAsync().ConfigureAwait(false);
-
-            using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
+            var stream = await _channel.SaveAsStreamAsync().ConfigureAwait(false);
+            await using (stream.ConfigureAwait(false))
             {
-                await stream.StreamImpl.CopyToAsync(fileStream).ConfigureAwait(false);
+                using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
```
Comment (kblok) [inline src/Playwright/Core/Artifact.cs:75]: You should be able to put these two usings at the same level

```
await using (stream.ConfigureAwait(false))
using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
{
    ...
}
```

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -25,12 +25,14 @@ jobs:
...
+            5.0.x
+            6.0.x
       - name: Install prerequisites and download drivers
         shell: bash
         run: ./build.sh --download-driver
       - name: Building
-        run: dotnet build ./src
+        run: dotnet build -f net5.0 ./src
       - name: Installing Browsers and dependencies
         run: dotnet run --project ./src/Playwright/Playwright.csproj -f net5.0 -- install --with-deps ${{ matrix.browser }}
```
Comment (kblok) [inline .github/workflows/tests.yml:37]: If we don't drop .NET 5, we should add the version to a matrix to test both versions. I would drop .NET 5.

File: src/Playwright/Core/Artifact.cs (hunk 5 above)
Comment (mxschmitt) [reply src/Playwright/Core/Artifact.cs:75]: linting rules are conflicting there thats why I put it that way: https://github.com/dotnet/roslyn-analyzers/issues/5712

File: .github/workflows/tests.yml (hunk 4 above)
Comment (mxschmitt) [reply .github/workflows/tests.yml:58]: we test on 'netcoreapp3.1' thats why its required I guess.

File: .github/workflows/tests.yml (hunk 6 above)
Comment (mxschmitt) [reply .github/workflows/tests.yml:37]: lets do it once .NET 5 reaches EOL or we follow-up earlier.

**Pull request merged**